packages/dashboard/src/utils/utils.ts

```diff
@@ -0,0 +1,79 @@
+import { getMessageBusPorts, PortsConfig } from "@truffle/dashboard-message-bus";
+import delay from "delay";
+import { JSONRPCRequestPayload } from "ethereum-protocol";
+import { promisify } from "util";
+import { INTERACTIVE_REQUESTS } from "./constants";
+import { BrowserProviderRequest, Request } from "./types";
+
+export const jsonToBase64 = (json: any) => {
+  const stringifiedJson = JSON.stringify(json);
+  const buffer = Buffer.from(stringifiedJson);
+  const base64 = buffer.toString("base64");
+
+  return base64;
+};
+
+export const base64ToJson = (base64: string) => {
+  const buffer = Buffer.from(base64, "base64");
+  const stringifiedJson = buffer.toString("utf8");
+  const json = JSON.parse(stringifiedJson);
+
+  return json;
+};
+
+// TODO: Replace these functions with those in the Dashboard Message Bus package
+// Only difference rn is that this uses browser WebSocket, while the other uses Node.js WebSocket
+export const connectToMessageBusWithRetries = async (port: number, retries = 1, tryCount = 1): Promise<WebSocket> => {
```

Comment:
This needs to be updated to support `host`; it looks like you did that for the dashboard-message-bus version, but missed here.

Reply:
Will be doing asap. Thank you so much